connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/WorkerCoordinator.java

```diff
@@ -257,13 +257,23 @@ public WorkerCoordinatorMetrics(Metrics metrics, String metricGrpPrefix) {
 
             Measurable numConnectors = new Measurable() {
                 public double measure(MetricConfig config, long now) {
-                    return assignmentSnapshot.connectors().size();
+		    if (assignmentSnapshot != null) {
```

Comment:
To simplify this, you could also just do `return assignmentSnapshot != null ? assignmentSnapshot.connectors().size() : 0.0;`